exporter/datadogexporter/internal/attributes/attributes_test.go

```diff
@@ -52,3 +53,53 @@ func TestTagsFromAttributesEmpty(t *testing.T) {
 
 	assert.Equal(t, []string{}, TagsFromAttributes(attrs))
 }
+
+func TestContainerTagsFromAttributes(t *testing.T) {
+	attributeMap := map[string]string{
+		conventions.AttributeContainerName:         "sample_app",
+		conventions.AttributeContainerImageTag:     "sample_app_image_tag",
+		conventions.AttributeK8SContainerName:      "kube_sample_app",
+		conventions.AttributeK8SReplicaSetName:     "sample_replica_set",
+		conventions.AttributeK8SDaemonSetName:      "sample_daemonset_name",
+		conventions.AttributeK8SPodName:            "sample_pod_name",
+		conventions.AttributeCloudProvider:         "sample_cloud_provider",
+		conventions.AttributeCloudRegion:           "sample_region",
+		conventions.AttributeCloudAvailabilityZone: "sample_zone",
+		conventions.AttributeAWSECSTaskFamily:      "sample_task_family",
+		conventions.AttributeAWSECSClusterARN:      "sample_ecs_cluster_name",
+		conventions.AttributeAWSECSContainerARN:    "sample_ecs_container_name",
+		"custom_tag":                               "example_custom_tag",
+		"":                                         "empty_string_key",
+		"empty_string_val":                         "",
+	}
+
+	containerMap := make(map[string]string)
+	containerTags := ContainerTagsFromAttributes(attributeMap)
+
+	for _, v := range strings.Split(containerTags, ",") {
```

Comment:
After you make the suggested changes, you won't need to deconstruct here because the new type `[]string` will be ordered and you can compare directly to the result.